### ISSUE TITLE
Should successful runs remove past `on-failure` comments?

### ISSUE THREAD

**Issue by gabrielfeo** [gradle/actions#176]
>Given I set `add-job-summary-as-pr-comment: 'on-failure'`,
>And 'foo' job has failed leaving a PR comment with the failure
>When a later run of 'foo' succeeds

Then should the failure comment be removed? From what I gathered, that's not implemented, so I'm opening a discussion about it. Repro of current behavior: https://github.com/gabrielfeo/develocity-api-kotlin/pull/233

In my opinion, it'd be best to either remove the comment or update it with an OK sign and the latest (successful) build scan. When the job succeeds, the failure comment becomes outdated info.

**Comment by bigdaz** [comment]
You're right we don't do anything to remove or modify previous job summary comments when submitting a new one. 
Likely the ideal default behaviour would be to replace any outdated PR comments with a simple link to the GitHub Actions build (removing the large summary). Although removing it would also be reasonable behaviour I guess.

**Comment by SimonMarquis** [comment]
Here is my attempt at implementing this:
- https://github.com/gradle/actions/pull/902

**Comment by SimonMarquis** [comment]
@bigdaz please take a look at the proposed PR when you have time 🙏
